### PR TITLE
Add Edit CSS menu item to appearance sub menu

### PR DESCRIPTION
**Pull request by tjcafferkey**
Fixes https://github.com/Automattic/wp-calypso/issues/50569

#### Changes proposed in this Pull Request:

* Adds "Edit CSS" menu item to the Appearance sub menu

#### Jetpack product discussion

N/A

#### Does this pull request change what data or activity we track or use?

No

#### Testing instructions:

**Simple Sites**

  - Apply this diff to your Sandbox `phab patch D58352`
  - Sandbox your host file and test sites for WP.com API (Ensure you are sandboxing a site with a Free plan and a Premium plan)
  - Force reconnection on your AutoProxxy to ensure you're using the Sandbox API
  - Load `your-free-site-id.wordpress.com` and hover over Appearance menu. You should see the "Edit CSS" menu item which takes you to the CSS nudge upgrade in Customizer.
  - Load `your-premium-site-id.wordpress.com` and hover over Appearance menu. You should see the "Edit CSS" menu item which takes you to the CSS editor in Customizer.

**Atomic Sites**

- Upload [Jetpack Beta](https://jetpack.com/download-jetpack-beta/) plugin
- Active this feature branch by going to Jetpack > Jetpack Beta menu item.
- Test there is no regression to Atomic sites, and also that there are no duplicate "Edit CSS" menu items.

### CODE REVIEW

File: projects/plugins/jetpack/modules/masterbar/admin-menu/class-wpcom-admin-menu.php [hunk 1]
```diff
@@ -200,6 +200,34 @@ public function add_upgrades_menu() {
...
+		parent::add_appearance_menu( $wp_admin_themes, $wp_admin_customize );
+
+		$themes_slug = $wp_admin_themes ? 'themes.php' : 'https://wordpress.com/themes/' . $this->domain;
+
+		if ( ! $wp_admin_customize ) {
+			$customize_slug = 'https://wordpress.com/customize/' . $this->domain;
+		} else {
+			// In case this is an api request we will have to add the 'return' querystring via JS.
+			$customize_slug = $this->is_api_request ? 'customize.php' : add_query_arg( 'return', rawurlencode( remove_query_arg( wp_removable_query_args(), wp_unslash( $_SERVER['REQUEST_URI'] ) ) ), 'customize.php' ); // phpcs:ignore WordPress.Security.ValidatedSanitizedInput.InputNotSanitized, WordPress.Security.ValidatedSanitizedInput.InputNotValidated
+		}
```
Comment (cpapazoglou) [inline projects/plugins/jetpack/modules/masterbar/admin-menu/class-wpcom-admin-menu.php:219]: All Simple sites should open the `$customize_slug = 'https://wordpress.com/customize/' . $this->domain;`. We can remove the conditional.

**Review by cpapazoglou — CHANGES_REQUESTED** [review]
Thanks for working on this, it works great.

We can remove the `$customize_slug` as Simple sites should always load customiser inside WordPress.com, which will make code simpler too.

File: projects/plugins/jetpack/changelog/add-edit-css-appearance-menu-item [hunk 2]
```diff
@@ -0,0 +1,4 @@
+Significance: enhancement
```
Comment (cpapazoglou) [inline projects/plugins/jetpack/changelog/add-edit-css-appearance-menu-item:1]:
```suggestion
Significance: minor
```

File: projects/plugins/jetpack/changelog/add-edit-css-appearance-menu-item [hunk 3]
```diff
@@ -0,0 +1,4 @@
...
+Type: added
```
Comment (cpapazoglou) [inline projects/plugins/jetpack/changelog/add-edit-css-appearance-menu-item:2]:
```suggestion
Type: enhancement
```

**Review by jeherve — APPROVED** [review]
This tests well for me. 👍

**Comment by tjcafferkey** [comment]
Deployed as part of r222552-wpcom

**Comment by jeherve** [comment]
Cherry-picked to `jetpack/branch-9.5` in e76d69957441b2167abc23a1ffdaf2ba816a438c

**Pull request merged**